redhat-appstudio/service-provider-integration-operator
# feat: SVPI-61 - vault deploy, init and configure

### What does this PR do?
vault deployment and configuration on minikube or openshift

 - minikube `make deploy_minikube`
 - openshift `make deploy`
 - script to init vault 
   - initialize vault and stores unseal keys into `spi-vault-keys` secret
   - setup kubernetes auth
   - setup audit log
   - setup `spi` kv secret engine
 - automatically unseal vault at pod startup
 - script for root access
 - generate new deployment yamls with `./hack/vault-generate-template.sh`
 - updates README

I've tested on minikube, openshift deployed with `make deploy`, openshift with infra-deployments


### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-61

### How to test this PR?

1. have `kubectl` configured for desired cluster, openshift or minikube
2. run `make deploy_minikube` or `make deploy` for openshift
3. Vault should be initialized and configured. You can go to Vault shell, run `vault status`. Should be initialized and unsealsed
4. To gain root access on vault pod, run `sh /vault/userconfig/scripts/root.sh`. This generates root token and login with it.
